### ISSUE TITLE
On Linux keyboard focus remains on owner window after calling ShowDialog

### ISSUE THREAD

**Issue by rprimora-pricer** [AvaloniaUI/Avalonia#13134]
## Describe the bug
If I call ShowDialog, keyboard focus remains on the owner window.

## To Reproduce

1. Create a new standard Avalonia project targeting Desktop
2. On the MainWindow that is supplied with Avalonia project template add a couple of input controls like textbox or checkbox.
3. Add a button that should open a new dialog and add a click event.
4. Add a new Avalonia Window (lets call it DialogWindow) and place some input controls on it like textbox or checkbox.
6. On MainWindow click event that you added call `await new DialogWindow().ShowDialog(this);`
7. Once the application starts position your focus in first input control of MainWindow just so it has focus.
8. Now click the button that opens up a dialog.
9. Now try to use tab to cycle through controls and observer how the focus remains on MainWindow when it should be in the DialogWindow.

## Expected behavior
Once a new Window is opened as a dialog it should have the keyboard focus.

## Environment
- OS: Ubuntu 22.04 LTS (Jammy Jellyfish) (64-bit) running on Virtual Box.
- Avalonia-Version: 11.0.4

**Comment by maxkatz6** [comment]
Related/duplicate https://github.com/AvaloniaUI/Avalonia/issues/2958

**Comment by kuiperzone** [comment]
I'm experiencing this with 11.3.9 on Wayland.

ShowDialog() does not disable Owner window, which still receives key events and tabbing means that controls are navigated on the window behind, not on the front dialog.

A work around is to disable the Owner.Window during the ShowDialog() process, and re-enable it on dialog close.
